### PR TITLE
[Prim] fix composite op map

### PR DESCRIPTION
**Pull request by cyber-pioneer**
### PR types
Bug fixes 

### PR changes
Others
### Describe
1. add map info of ops from yaml description to composite rules: mean, layer_norm, dropout
2. improvement of usability.
  a. support output map from origin ops to composite rule: 
        When user defines some composite rule of one op，its inputs and outputs keep the same as yaml.
  b. support composite op call other composite op:
        When user defines some composite rule, the ops which have been defined composite rule can also be called.
  c. add log of name list of composite ops:
       The ops which has been replaced by composite rules will be printed.

### CODE REVIEW

File: python/paddle/incubate/autograd/primx.py [hunk 1]
```diff
@@ -596,19 +598,38 @@ def expand_nested_list(xs):
...
+                    f'but len(orig_outs) = {len(orig_outs)} and len(new_outs) = {len(new_outs)}'
+                )
                 for orig_out, new_out in zip(
-                    expand_nested_list(get_output_var_list(op)),
-                    expand_nested_list(as_tensors(lower_fn(op, *input_args))),
+                    orig_outs,
+                    new_outs,
                 ):
                     if new_out is not None:
+                        if orig_out.shape and new_out.shape:
```
Comment (JiabinYang) [inline python/paddle/incubate/autograd/primx.py:633]: need other check like vartype and dtype

File: python/paddle/incubate/autograd/utils.py [hunk 2]
```diff
@@ -218,6 +227,37 @@ def get_output_var_list(op):
         ]
 
 
+def get_output_vars_from_comosite(op):
```
Comment (JiabinYang) [inline python/paddle/incubate/autograd/utils.py]: bad name, maybe call it `rearrange_output_for_composite`

File: python/paddle/incubate/autograd/utils.py [hunk 3]
```diff
@@ -218,6 +227,37 @@ def get_output_var_list(op):
...
+    """origin op outputs must be mapped into outputs of composite rule."""
```
Comment (JiabinYang) [inline python/paddle/incubate/autograd/utils.py]: more comments for it, its' some like a trick

File: python/paddle/incubate/autograd/utils.py [hunk 4]
```diff
@@ -202,7 +206,12 @@ def prepare_python_api_arguments(op):
         else:
             phi_name = op.type
         inputs, attrs = _get_args_values(op, phi_name)
-        res = [get_var_block(op.block, op.input(n)) for n in inputs]
+        res = []
+        for item in inputs:
+            if item in op.input_names:
+                res.append(get_var_block(op.block, op.input(item)))
+            else:
+                res.append(None)
```
Comment (JiabinYang) [inline python/paddle/incubate/autograd/utils.py:216]: This maybe not safe here, but keep it and add a todo to reminds

File: python/paddle/incubate/autograd/utils.py [hunk 5]
```diff
@@ -182,8 +183,11 @@ def _get_args_values(op, phi_name):
                 "attrs" in op_content.keys()
                 and arg_name in op_content["attrs"].keys()
             ):
-                attrs.append(op.attr(op_content["attrs"][arg_name]))
-            attrs.append(op.attr(arg_name))
+                arg_name = op_content["attrs"][arg_name]
+            if arg_name not in op.attr_names:
```
Comment (JiabinYang) [inline python/paddle/incubate/autograd/utils.py:188]: same not safe

**Review by JiabinYang — COMMENTED** [review]
some comments

File: python/paddle/incubate/autograd/primx.py (hunk 1 above)
Comment (cyber-pioneer) [reply python/paddle/incubate/autograd/primx.py:633]: some attrs may be not the same, and also right.
todo has been added

File: python/paddle/incubate/autograd/utils.py (hunk 2 above)
Comment (cyber-pioneer) [reply python/paddle/incubate/autograd/utils.py]: done

File: python/paddle/incubate/autograd/utils.py (hunk 3 above)
Comment (cyber-pioneer) [reply python/paddle/incubate/autograd/utils.py]: done

File: python/paddle/incubate/autograd/utils.py (hunk 5 above)
Comment (cyber-pioneer) [reply python/paddle/incubate/autograd/utils.py:188]: note has been added

File: python/paddle/incubate/autograd/utils.py (hunk 4 above)
Comment (cyber-pioneer) [reply python/paddle/incubate/autograd/utils.py:216]: note has been added

File: paddle/phi/api/yaml/op_compat.yaml [hunk 6]
```diff
@@ -145,6 +145,13 @@
     variance : Variance
     scale : Scale
     bias : Bias
+  outputs :
+    out : Y
+    mean_out: MeanOut
+    variance_out: VarianceOut
+    saved_mean: SavedMean
+    saved_variance: SavedVariance
+    reserve_space: ReserveSpace
```
Comment (zyfncg) [inline paddle/phi/api/yaml/op_compat.yaml:154]: 参数多的情况可以用`{}`的形式配置，能减少些文件行数

Comment (cyber-pioneer) [reply paddle/phi/api/yaml/op_compat.yaml:154]: 下个pr修改格式

**Review by zyfncg — APPROVED** [review]

**Review by JiabinYang — APPROVED** [review]
LGTM

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**